metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -330,13 +321,14 @@ private void open() throws MetaException {
         LOG.info("Trying to connect to metastore with URI " + store);
         try {
           try {
-            InetSocketAddress server = new InetSocketAddress(store.getHost(), store.getPort());
-            NetUtils.connect(socket, server, 1000 * clientSocketTimeout);
-            transport = new TSocket(socket);
-          } catch (TTransportException e) {
-              LOG.error("Couldn't create client transport to HiveMetastore(" 
-                + store.getHost() + ":" + store.getPort() + ")", e);
-              throw new MetaException(e.toString());
+            TSocket tsocket = new TSocket(store.getHost(), store.getPort(), 1000 * clientSocketTimeout);
+            if (hiveHost != null && !hiveHost.isEmpty())  {
+              socket = tsocket.getSocket();
+              InetAddress localAddr = NetUtils.getLocalInetAddress(hiveHost);
+              if (localAddr != null)
```

Comment:
I still think this needs an 'else' clause to log the unexpected behavior of localAddr being null.  It's not going to work right in this case, will it?   If it does work right when localAddr is null, why do you need to do any of this?